### PR TITLE
Prevent context deadline error propagation in GetCommitsInfo

### LINKED ISSUE #20329: Context deadline exceeded
### Description

One of the mirrored project, [LLVM/LLVM-project](https://github.com/llvm/llvm-project) seems to have a problem that anytime I try to open it in gitea server, a 500 error is return with context deadline exceeded which seems to suggest that I am hitting a Time to Live timer limit when attempting to open the page for LLVM mirrored project. I've tried asking about it on matrix chat, checked FAQ, and other google search for this but I was not able to find a solution for this yet.

The local gitea server is running on Proxmox server with about 8 cores, 32 gb of RAM, and 1 TB of storage in LVM Ext4 file system. I also can't reproduce this on https://try.gitea.io since I am unable to get an activation email to try replicating the issue.

### Gitea Version

1.17.0-rc1

### Can you reproduce the bug on the Gitea demo site?

No

### Log Gist

https://gist.github.com/CLanguagePurist/66337b4af241f89f7f40a5dcc4d40946

### Screenshots

giteaerror

### Git Version

2.25.1

### Operating System

Ubuntu 20.04.4 LTS

### How are you running Gitea?

Using executable download for Linux from tagged release for 1.17.0-rc1, gitea-1.17.0-rc1-linux-amd64 from [this page](https://github.com/go-gitea/gitea/releases/tag/v1.17.0-rc1) and have done SHA256 checksum comparison and they matched with gitea-1.17.0-rc1-linux-amd64.sha256. I have granted this executable privilege to operates on port 80 and 443 and this server is **NOT** visible to the public internet and required firewall whitelisted address on the LAN to access the server IE my development machine which also runs Linux.

### Database

PostgreSQL

### PR DESCRIPTION
**Pull request by zeripath**
Although `WalkGitLog` tries to test for `context.DeadlineExceeded`
there is a small chance that the error will propagate to the reader
before it is recognised. This will cause the error to propagate up to
`renderDirectoryFiles` and cause a http status 500.

Here we check that the error passed is a `DeadlineExceeded` or if it
could be unwrapped to one.

Fix #20329

Signed-off-by: Andrew Thornton <art27@cantab.net>

### CODE REVIEW

**Review by 6543 — APPROVED** [review]

File: modules/git/log_name_status.go [hunk 1]
```diff
@@ -354,7 +356,7 @@ heaploop:
 		}
 		current, err := g.Next(treepath, path2idx, changed, maxpathlen)
 		if err != nil {
-			if err == context.DeadlineExceeded {
+			if err == context.DeadlineExceeded || errors.Unwrap(err) == context.DeadlineExceeded {
```
Comment (Gusted) [inline modules/git/log_name_status.go]: Isn't `error.Is` made for this purpose?

Comment (zeripath) [reply modules/git/log_name_status.go]:
```suggestion
			if errors.Is(err, context.DeadlineExceeded) {
```

Comment (zeripath) [reply modules/git/log_name_status.go]: We have been burnt several times before with errors.Is - in this case - after having looked carefully at the code it should be correct.

**Review by Gusted — APPROVED** [review]

**Comment by 6543** [comment]
:robot:

**Comment by 6543** [comment]
-> #20361

**Pull request merged**